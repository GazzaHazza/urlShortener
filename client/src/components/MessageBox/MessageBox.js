import React, { Component } from "react";
import PropTypes from "prop-types";
import "./MessageBox.css";
class MessageBox extends Component {
  render() {
    const { message, hasError, hasAdded, shortUrl } = this.props;
    const className = hasError && !hasAdded ? "--error" : "--added";
    return (
      <div className={`message message${className}`}>
        {message}
        <a
          href={shortUrl}
          target="_blank"
          className={`message__link message__link${className}`}
        >
          {shortUrl}
        </a>
      </div>
    );
  }
}

MessageBox.propTypes = {
  hasError: PropTypes.bool,
  hasAdded: PropTypes.bool,
  shortUrl: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};
MessageBox.defaultProps = {
  hasError: false,
  hasAdded: false
};
export default MessageBox;
