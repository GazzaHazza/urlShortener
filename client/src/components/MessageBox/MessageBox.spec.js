import React from "react";
import MessageBox from "./MessageBox";
import { shallow } from "enzyme";

it("renders correctly with error class", () => {
  const wrapper = shallow(
    <MessageBox
      message="hello"
      hasError={true}
      hasAdded={false}
      shortUrl="www.hello.com/1"
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly with added class", () => {
  const wrapper = shallow(
    <MessageBox
      message="hello"
      hasError={false}
      hasAdded={true}
      shortUrl="www.hello.com/1"
    />
  );
  expect(wrapper).toMatchSnapshot();
});
