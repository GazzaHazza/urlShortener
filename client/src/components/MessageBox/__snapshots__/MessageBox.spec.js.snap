// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with added class 1`] = `
<div
  className="message message--added"
>
  hello
  <a
    className="message__link message__link--added"
    href="www.hello.com/1"
    target="_blank"
  >
    www.hello.com/1
  </a>
</div>
`;

exports[`renders correctly with error class 1`] = `
<div
  className="message message--error"
>
  hello
  <a
    className="message__link message__link--error"
    href="www.hello.com/1"
    target="_blank"
  >
    www.hello.com/1
  </a>
</div>
`;
